lib/connectionConfig/iothubConnectionConfig.ts

```diff
@@ -76,23 +77,23 @@ export namespace IotHubConnectionConfig {
    * @param {ConnectionConfig} config The connection config to be validated.
    */
   export function validate(config: IotHubConnectionConfig): void {
-    if (!config || (config && typeof config !== "object")) {
-      throw new Error("'config' is a required parameter and must be of type: 'object'.");
-    }
-    if (!config.hostName || (config.hostName && typeof config.hostName !== "string")) {
-      throw new Error("'hostName' is a required property of the ConnectionConfig.");
-    }
-    if (!config.entityPath || (config.entityPath && typeof config.entityPath !== "string")) {
-      throw new Error("'entityPath' is a required property of the ConnectionConfig.");
-    }
-    if (!config.sharedAccessKeyName || (config.sharedAccessKeyName && typeof config.sharedAccessKeyName !== "string")) {
-      throw new Error("'sharedAccessKeyName' is a required property of the ConnectionConfig.");
-    }
-    if (!config.sharedAccessKey || (config.sharedAccessKey && typeof config.sharedAccessKey !== "string")) {
-      throw new Error("'sharedAccessKey' is a required property of the ConnectionConfig.");
-    }
-    if (config.deviceId != undefined && typeof config.deviceId !== "string") {
-      throw new Error("'deviceId' must be of type 'string'.");
+    throwTypeErrorIfParameterMissing("config", config);
+    throwTypeErrorIfParameterTypeMismatch("config", config, "object");
+
+    throwTypeErrorIfParameterMissing("hostName", config.hostName);
+    config.hostName = String(config.hostName);
```

Comment:
How come we don't need the hostname check in `ConnectionConfig.validate()`?

Reply:
Good catch!
They must have forgotten the hostname check here.